### PR TITLE
(PE-17621) Beaker answers generates meep node roles

### PR DESCRIPTION
**Pull request by jpartlow**
Generate node_roles in 2017.1+ pe_conf ￼…
MEEP 2.0 introduces a node_roles hash in pe.conf for classification.
This supercedes using the puppet_enterprise host parameters as a de
facto classification.  The Version20171 class maps existing beaker role
aliases to the new role strings used for classification in pe.conf.

So, from a MEEP 1.0 version pe.conf such as:

  {
    "puppet_enterprise::puppet_master_host": "master.net"
    "puppet_enterprise::puppetdb_host": "puppetdb.net"
    "puppet_enterprise::console_host": "console.net"
  }

You now have a 2.0 version that looks like:

  {
    "node_roles": {
      "pe_role::split::primary_master": ["master.net"]
      "pe_role::split::puppetdb": ["puppetdb.net"]
      "pe_role::split::console": ["console.net"]
    }
  }

See the pe_manager and pe_infrastructure modules for more details.

### CODE REVIEW

**Comment by jpartlow** [comment]
Requires puppetlabs/puppetlabs-puppet_enterprise#978 and puppetlabs/puppetlabs-pe_install#88 and puppetlabs/puppetlabs-pe_manager#89 and puppetlabs/puppetlabs-pe_infrastructure#11

File: spec/beaker-answers/versions/version20171_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+require 'spec_helper'
+require 'json'
+require 'pry-byebug'
```
Comment (demophoon) [inline spec/beaker-answers/versions/version20171_spec.rb]: Was this suppose to be included?

Comment (jpartlow) [inline spec/beaker-answers/versions/version20171_spec.rb]: Nope! Thanks, let me fix that up.

**Comment by johnmccabe** [comment]
Frankenbuild mono/split with smoke look good.

**Pull request merged**